xds/src/main/java/com/linecorp/centraldogma/xds/internal/XdsResourceWatchingService.java

```diff
@@ -84,26 +89,36 @@ protected void init() {
             watchingGroups.add(groupName);
             final Revision normalizedRevision = repository.normalizeNow(Revision.HEAD);
             logger.info("Creating xDS resources from {} at revision: {}", groupName, normalizedRevision);
-            final Map<String, Entry<?>> entries = repository.find(normalizedRevision, pathPattern()).join();
-            for (Entry<?> entry : entries.values()) {
-                if (entry.type() != EntryType.JSON || !entry.hasContent()) {
-                    continue;
+            final CompletableFuture<Map<String, Entry<?>>> findFuture =
+                    repository.find(normalizedRevision, pathPattern());
+            futures.add(findFuture.handleAsync((entries, cause) -> {
+                // Executed by the repository worker thread.
+                if (cause != null) {
+                    logger.warn("Unexpected exception while finding {} at revision: {}",
```

Comment:
Question) It seems like this behavior has changed from failing the startup to just logging a warning.
When restarting, are we expected to check for this log? or would just explicitly throwing a `RuntimeException` make more sense?

Reply:
I thought it was okay to skip the repository but it was probably not a good idea.
Let me re-throw the exception in this case. 😉